rt/transports/http/src/main/java/org/apache/cxf/transport/http/HttpClientHTTPConduit.java

```diff
@@ -195,14 +207,22 @@ RefCount<HttpClient> computeIfAbsent(final boolean shareHttpClient, final HTTPCl
 
             // Do not share if it is not allowed for the conduit or cache capacity is exceeded
             if (!shareHttpClient || clients.size() >= MAX_SIZE) {
-                return new RefCount<HttpClient>(supplier.get(), policy, clientParameters, () -> { }).acquire();
+                return new RefCount<HttpClient>(supplier.get(), policy, clientParameters, () -> { });
             }
 
             lock.lock();
             try {
                 for (final RefCount<HttpClient> p: clients) {
-                    if (cpc.equals(p.policy(), policy) && p.clientParameters().equals(clientParameters)) {
-                        return p.acquire();
+                    try {
```

Comment:
I think this is fine as well, the outer `try` in only to fail fast (we usually only have on client per that matches  `cpc.equals(p.policy(), policy) && p.clientParameters().equals(clientParameters))` , thanks @ffang !